### DISCUSSION TITLE
Changing the dynamic-title

### DISCUSSION THREAD

**Discussion by SimoneFelici** [cotes2020/jekyll-theme-chirpy#1751] [Q&A] [answered]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Generated from `chirpy-starter`

### Description

Hi, I want to change the dynamic-title of one of the tabs:
From this:
![image](https://github.com/cotes2020/jekyll-theme-chirpy/assets/57272391/67cd2058-0d77-414b-ac3c-921e9d9ebc8a)
To this:
![image](https://github.com/cotes2020/jekyll-theme-chirpy/assets/57272391/23a4829f-cca8-4ed3-ba59-a6a5f5f6f705)

The problem is that I dont want to influence the nav item one:
![image](https://github.com/cotes2020/jekyll-theme-chirpy/assets/57272391/fd454612-5b8d-4cac-9458-641fbd5a2959)
Thanks

### Operations you have already tried

I've tried to search in the css, but I didn't find the option.

**Comment by huanyushi** [comment, marked as answer]
Your question is quite confusing, I'm not sure what you want to change.

Do you mean to achieve this effect?

![image](https://github.com/cotes2020/jekyll-theme-chirpy/assets/100574401/474d50c4-a580-4dc4-9595-00e54a37e56d)

---
**Update**

@SimoneFelici Just add title in front matter, like this:

```markdown
---
title: ~$ about
---
```

![image](https://github.com/cotes2020/jekyll-theme-chirpy/assets/100574401/87dcb3a9-b8c4-41ee-ae6f-e965f32544d3)

**Reply by SimoneFelici** [reply]
Yeah, that's what I wanted to change, but I want to change only the about page and not the others, like ARCHIVES etc...
Also it would be great if there was a space between the $ and the A, and that the A was lowercase.

**Reply by huanyushi** [reply]
@SimoneFelici I've edited my original answer, which I think solves your problem.

**Reply by SimoneFelici** [reply]
yeah, that makes sense😅
thank you

**Reply by SimoneFelici** [reply]
Oh, wait, I've noticed that if you do it like that, it also changes the title in the navbar:
![image](https://github.com/cotes2020/jekyll-theme-chirpy/assets/57272391/3ae34c3a-4f6f-4d00-b6fc-091bc8d7d424)
Is there a way to do that without changing the navbar?

**Reply by huanyushi** [reply]
@SimoneFelici In my testing, this did not change the sidebar title, as shown in the screenshot in my answer. Maybe you made other mistakes?

**Reply by SimoneFelici** [reply]
Mh, you are right, I don't know, I actually changed about.md inside _tabs, but I dont know where I should put the title:
![image](https://github.com/cotes2020/jekyll-theme-chirpy/assets/57272391/04735741-1e8b-4da7-83a0-bdd5132b741c)
Do I need to edit index.html under _site/about/
?

**Reply by SimoneFelici** [reply]
Oh, I have found the problem, that's because I changed the name of the .md file to whoami, if I rename it about it works like your site.
The strange thing is, if I put whoami it takes the
```
---
title:
---
```
I don't know how to change that if it's possible

**Reply by huanyushi** [reply]
@SimoneFelici I don't understand what you mean, why is there something strange happening? Can you be more clear? These are just simple settings, and like in my screenshot, everything works perfectly.

**Reply by SimoneFelici** [reply]
Yeah, sorry I will be more specific.
Your md file inside _tabs is called about.md
And you changed the title using
```
title: ~$ whoami
```
And now you have in the navbar "ABOUT" 
And in the page "~$ whoami"
And that's basically what I want, but I have previously changed the md file in "whooami.md" instead of "about.md"
So in the navbar I have "WHOAMI"
But for a strange thing, maybe is supposed to do that, but I don't want that.
If you have the file called whoami.md and the title called ~$ whoami 
It changed the navbar and the page.
I really don't know why, because as you can see if the file is called about it doesn't change the navbar into ~$ whoami.
Hope to clear things up, it's not that big of a deal, just want to edit that

**Reply by huanyushi** [reply]
@SimoneFelici OK, I got it. I think the relevent code is [here](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/_layouts/page.html#L10-L13). 

```liquid
{% assign title = site.data.locales[lang].tabs[tab_key] | default: page.title %}
```

If program find `site.data.locales[lang].tabs[tab_key]` in lang.yml (e.g. `_data/locales/en.yml`), title will be set to this value. If not found, it will point to the title you filled in. So I think the easiest way is to add the following code in [`_data/locales/en.yml`](https://github.com/cotes2020/jekyll-theme-chirpy/blob/90a4cc76bb4a8ee843361b5024a6f7d8f924461c/_data/locales/en.yml#L10-L17) (if you set `lang: en` in `config.yml`):

```diff
# The tabs of sidebar
tabs:
  # format: <filename_without_extension>: <value>
  home: Home
  categories: Categories
  tags: Tags
  archives: Archives
  about: About
+ whoami: whoami
```

Then you create a file `_tabs/whoami.md`, and set the title in front matter:

```markdown
---
# the default layout is 'page'
icon: fas fa-info-circle
order: 4
title: ~$ whoami
---
```

This way you can achieve the functionality you want.

<p align="center">
<img src="https://github.com/cotes2020/jekyll-theme-chirpy/assets/100574401/94cb68fb-59c3-4860-bf2e-cc1c058ed8e0" width="50%">
</p>

**Reply by huanyushi** [reply]
@SimoneFelici Another solution is just creating a new layout, e.g. `_layouts/whoami.html`

```diff
---
layout: default
---

{% include lang.html %}

<article class="px-1">
  {% if page.layout == 'page' or page.collection == 'tabs' %}
    {% assign tab_key = page.title | downcase %}
-    {% assign title = site.data.locales[lang].tabs[tab_key] | default: page.title %}
+    {% assign title = '~$ whoami' %}
    <h1 class="dynamic-title">
      {{ title }}
    </h1>
    <div class="content">
      {{ content }}
    </div>
  {% else %}
    {{ content }}
  {% endif %}
</article>
```

where I show the difference from `_layouts/page.html` and I specify the title as you want `~$ whoami`. Then back to `_tabs/whoami.md`, you just need to change the front matter as:

```markdown
---
# the default layout is 'page'
layout: whoami
icon: fas fa-info-circle
order: 4
---
```

Both methods can achieve the functions you want, you can choose any.

**Reply by SimoneFelici** [reply]
Yeah that works, thank you very much, I didn't had those options because they don't come with the starter, so I had to add them manually.
